symbolic-debuginfo/src/base.rs

```diff
@@ -602,21 +602,27 @@ pub type DynIterator<'a, T> = Box<dyn Iterator<Item = T> + 'a>;
 ///  - Read headers of compilation units (compilands) to resolve cross-unit references.
 ///
 /// [`ObjectLike::debug_session`]: trait.ObjectLike.html#tymethod.debug_session
-pub trait DebugSession {
+pub trait DebugSession<'session> {
```

Comment:
Usually, we don't need the `'data` lifetime here, but do we need the `'object` one?

Reply:
On taking a second look, I think we can get away without, since we can still bind it in the associated types on the concrete implementation. In that case, `'object: 'session` would need to be fulfilled.

Reply:
Yes, I think that should be enough.

Reply:
What if we do the same for the `ObjectLike` trait, to only have the `'object` lifetime there?

Reply:
To return references tied to the objects lifetime?

Reply:
Correct. This would give us the benefit of not having to clone in `PdbObject::symbols`, and I think that use cases where you need the `'data` life time but cannot hold the objects are very scarce in hindsight.